src/backend/CassandraBackend.cpp

```diff
@@ -695,6 +733,24 @@ CassandraBackend::fetchNFTTransactions(
     return {txns, {}};
 }
 
+bool
+CassandraBackend::fetchIsMigrationExecuted(
+    std::string const& migration,
+    boost::asio::yield_context& yield) const
+{
+    CassandraStatement query{selectClioMigration_};
+    query.bindNextString(migration);
+    return executeAsyncRead(query, yield).hasResult();
+}
+
+void
+CassandraBackend::writeMigrationReceipt(std::string const& migration) const
+{
+    CassandraStatement statement(insertClioMigration_);
+    statement.bindNextString(migration);
+    executeSyncWrite(statement);
```

Comment:
perhaps this should be async too?

Reply:
I'm not sure anything should be async when performing migrations..

Reply:
This makes sense to be synchronous. It's a single write, done at the end of the migration. I'd imagine you also want the semantics of knowing that the receipt is viewable by all readers.